app/controllers/image-set.js

```diff
@@ -0,0 +1,73 @@
+import Ember from 'ember';
+import {imageTypes, genders} from 'lion-guardians/utils/units';
+
+export default Ember.Controller.extend({
+  imageTypes: imageTypes,
+  genders: genders,
+  uploadImageType: imageTypes[0],
+  uploadIsPublic: false,
+  isEditing: false,
+
+  organizations: function() {
+    return this.store.findAll('organization');
+  }.property(),
+
+  actions: {
+    addImage: function(upload){
+      var url = upload.get('url'),
+          isPublic = this.get('uploadIsPublic'),
+          imageType = this.get('uploadImageType'),
+          model = this.get('model');
+
+      model.addImage(url, isPublic, imageType);
+
+      Ember.run.later(this, function(){
+        // delete upload so it disappears from upload window
+        upload.deleteRecord();
+      }, 3000);
+    },
+
+    makeMainImage: function(image) {
+      this.set('model.mainImage', image);
+    },
+
+    deleteImage: function(image) {
+      var imageSet = this.get('model');
+      imageSet.get('images').removeObject(image);
+      if (imageSet.get('mainImage') === image) {
+        imageSet.set('mainImage', null);
+      }
+      image.destroy();
```

Comment:
Does this fire an ajax call? I guess it does not because it hasn't been persisted yet?